lib/semantics/check-omp-structure.cc

```diff
@@ -385,6 +385,24 @@ void OmpStructureChecker::Enter(const parser::OpenMPBlockConstruct &x) {
         OmpClause::IF, OmpClause::FINAL, OmpClause::PRIORITY};
     SetContextAllowedOnce(allowedOnce);
   } break;
+  // 2.10.4 target-clause -> if-clause |
+  //                         device-clause |
+  //                         private-clause |
+  //                         firstprivate-clause |
+  //                         map-clause |
+  //                         is-device-ptr-clause |
+  //                         defaultmap-clause |
+  //                         nowait-clause |
+  //                         depend-clause
+  case parser::OmpBlockDirective::Directive::Target: {
+    PushContext(beginDir.source, OmpDirective::TARGET);
+    OmpClauseSet allowed{OmpClause::IF, OmpClause::PRIVATE, OmpClause::MAP,
+        OmpClause::DEPEND, OmpClause::FIRSTPRIVATE, OmpClause::IS_DEVICE_PTR};
```

Comment:
Nit. Can you re-order the allowed set to match the order of the clauses above?

Reply:
Done, they are now the same order as the grammar